### DISCUSSION TITLE
Reusing texture atlases

### DISCUSSION THREAD

**Discussion by hamdav** [bevyengine/bevy#3740] [Q&A]
Hello!

I was wondering if you should reuse already loaded texture atlases when you spawn new entities. For example, following the [sprite sheet example](https://github.com/bevyengine/bevy/blob/latest/examples/2d/sprite_sheet.rs), if I want to spawn new animated sprites while the game is running, should I do this:

    let texture_handle = asset_server.load("textures/rpg/chars/gabe/gabe-idle-run.png");
    let texture_atlas = TextureAtlas::from_grid(texture_handle, Vec2::new(24.0, 24.0), 7, 1);
    let texture_atlas_handle = texture_atlases.add(texture_atlas);

again? Should I save the handle in a resource? Something like a HashMap mapping names to handles. Or is there some other way of getting a previously loaded textures?

**Comment by automeowtion** [comment]
https://bevy-cheatbook.github.io/features/assets.html

**Comment by patheros** [comment]
I have the same question as the original poster, but the above link is 404. Is there an updated link or does someone else know the answer?

**Comment by bugsweeper** [comment]
Short answer is no. AssetServer returns handles immediatly, but texture (or other resource) behind this handle will be available later, you can reuse that handles multiple times. This is very usable when you can do fast setup, while application uses lazy loading. You should read [this article](https://bevy-cheatbook.github.io/assets/handles.html) for more info

**Comment by patheros** [comment]
Thank you for your quick reply. I read through the article you sent and that gives me a better understanding of the asset server and what it is doing.

However I still have a question about the TextureAtlas and its construction. The code in [this example](https://github.com/bevyengine/bevy/blob/latest/examples/2d/sprite_sheet.rs) calls `TextureAtlass::from_grid` in the setup. But in the case where you are spawning many of these sprites, should the `TextureAtlas` be saved somewhere so it doesn't have to be reconstructed each time?

**Reply by bugsweeper** [reply]
[Assets::add](https://docs.rs/bevy/latest/bevy/asset/struct.Assets.html#method.add) also returns handle, as you see in example you should add TextureAtlass and then use it's handle. So you can add TextureAtlass once, and then use it's handle every where with same principle

**Reply by miketwenty1** [reply]
Is the primary method for saving these handles to put them in Resources?
